src/renderer/mesh_storage.zig

```diff
@@ -859,6 +853,55 @@ pub fn updateMeshes(targetTime: i64) void { // MARK: updateMeshes()
 	}
 }
 
+fn batchUpdateBlocks() void {
+	var lightRefreshList = main.List(*ChunkMesh).init(main.stackAllocator);
+	defer lightRefreshList.deinit();
+
+	var regenerateMesh: main.ListUnmanaged(*ChunkMesh) = .{};
+	defer regenerateMesh.deinit(main.stackAllocator);
+
+	// First of all process all the block updates:
+	while(blockUpdateList.dequeue()) |blockUpdate| {
+		const pos = chunk.ChunkPosition{.wx = blockUpdate.x, .wy = blockUpdate.y, .wz = blockUpdate.z, .voxelSize = 1};
+		if(getMeshAndIncreaseRefCount(pos)) |mesh| {
+			const result = mesh.updateBlock(blockUpdate.x, blockUpdate.y, blockUpdate.z, blockUpdate.newBlock, &lightRefreshList);
+			switch(result) {
+				.noChange => {
+					mesh.decreaseRefCount();
+					continue;
+				},
+				.regenerateMesh => {
+					var found = false;
+					for(regenerateMesh.items) |other| {
+						if(other.pos.equals(mesh)) {
+							mesh.decreaseRefCount();
+							found = true;
+							break;
+						}
+					}
+					if(!found) regenerateMesh.append(main.stackAllocator, mesh);
```

Comment:
Zig supports breaking from arbitrary blocks, which in my opinion is more readable than bools.
```suggestion
				.regenerateMesh => blk: {
					for(regenerateMesh.items) |other| {
						if(other.pos.equals(mesh)) {
							mesh.decreaseRefCount();
							break :blk;
						}
					}
					regenerateMesh.append(main.stackAllocator, mesh);
```